### PR TITLE
store: atomically committing and changing storage.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @tiancaiamao

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -668,23 +668,26 @@ impl Peer {
         peer_storage::save_applied_index(&wb, self.region_id, index)
```
Comment (ngaut) [inline src/raftstore/store/peer.rs]: De we need to make sure all of the writes are atomic

Comment (siddontang) [inline src/raftstore/store/peer.rs]: here just write to WriteBatch, no need atomic.

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**